src/lib/coda_metrics/coda_metrics.ml

```diff
@@ -455,6 +456,27 @@ module Transition_frontier_controller = struct
     Counter.v "breadcrumbs_built_by_builder" ~help ~namespace ~subsystem
 end
 
+let dump () : Yojson.Safe.json =
+  let f {MetricInfo.name; metric_type= _; help= _; label_names} samples acc =
+    (*
+    let labels =
+      List.map label_names ~f:(fun label_name ->
+        (label_name, Option.value ~default:"NONE" (LabelSetMap.find (label_name :> string) samples)))
+    in
+    (name, labels)
+    *)
+    let value_opt =
+      let open Option.Let_syntax in
+      let open Sample_set in
+      let%map samples =
+        LabelSetMap.find (label_names :> string list) samples
+      in
+      Float.to_string (List.hd_exn samples).value
+    in
+    ((name :> string), `String (Option.value ~default:"NONE" value_opt)) :: acc
+  in
+  `Assoc (MetricFamilyMap.fold f CollectorRegistry.(collect default) [])
```

Comment:
I’m not familiar with the metrics stuff, but should we be using a histogram here? Could you explain what this is doing as is?